src/sentry/replays/query.py

```diff
@@ -183,108 +187,32 @@ def query_replays_count(
-        _grouped_unique_scalar_value(column_name="device_family"),
-        _grouped_unique_scalar_value(column_name="device_model"),
-        _grouped_unique_scalar_value(column_name="sdk_name"),
-        _grouped_unique_scalar_value(column_name="sdk_version"),
-        # Flatten array of arrays.
-        Function(
-            "groupArrayArray",
-            parameters=[Column("tags.key")],
-            alias="tk",
-        ),
-        Function(
-            "groupArrayArray",
-            parameters=[Column("tags.value")],
-            alias="tv",
-        ),
-        Function(
-            "arrayMap",
-            parameters=[
-                Lambda(
-                    ["trace_id"],
-                    _strip_uuid_dashes("trace_id", Identifier("trace_id")),
-                ),
-                Function(
-                    "groupUniqArrayArray",
-                    parameters=[Column("trace_ids")],
-                ),
-            ],
-            alias="traceIds",
-        ),
-        Function(
-            "arrayMap",
-            parameters=[
-                Lambda(["error_id"], _strip_uuid_dashes("error_id", Identifier("error_id"))),
-                Function(
-                    "groupUniqArrayArray",
-                    parameters=[Column("error_ids")],
-                ),
-            ],
-            alias="errorIds",
-        ),
-        # Aggregations.
-        Function("min", parameters=[Column("replay_start_timestamp")], alias="startedAt"),
-        Function("max", parameters=[Column("timestamp")], alias="finishedAt"),
-        Function(
-            "dateDiff",
-            parameters=["second", Column("startedAt"), Column("finishedAt")],
-            alias="duration",
-        ),
-        Function(
-            "groupArray",
-            parameters=[Function("tuple", parameters=[Column("segment_id"), Column("urls")])],
-            alias="agg_urls",
-        ),
-        _sorted_aggregated_urls(Column("agg_urls"), "urls_sorted"),
-        Function("count", parameters=[Column("segment_id")], alias="countSegments"),
-        Function(
-            "uniqArray",
-            parameters=[Column("error_ids")],
-            alias="countErrors",
-        ),
-        Function(
-            "notEmpty",
-            parameters=[Function("groupArray", parameters=[Column("is_archived")])],
-            alias="isArchived",
-        ),
-        _activity_score(),
-    ]
+    if not fields:
+        return QUERY_ALIAS_COLUMN_MAP.values()
+
+    unique_fields = set(fields)
+
+    # Select fields used for filtering.
+    #
+    # These fields can filter a query without being selected.  However, if we did not select these
+    # values we could not reuse those columns which are expensive to calculate.  This coupled with
+    # the complexity of dependency management means we filter these columns manually in the final
+    # output.
+    for fltr in search_filters:
+        if isinstance(fltr, SearchFilter):
```

Comment:
why do we this `isinstance` check here?

Reply:
Updated the type signature.  This list can contain strings and ParenExpressions.